lib/membrane_live/event.ex

```diff
@@ -165,13 +164,33 @@ defmodule MembraneLive.Event do
     Engine.accept_peer(rtc_engine, peer.id)
     :ok = Engine.add_endpoint(rtc_engine, endpoint, peer_id: peer.id, node: peer_node)
 
+    state = update_in(state, [:playlist_idls], fn items -> [%{peer_id: peer.id} | items] end)
+
     {:noreply, state}
   end
 
   @impl true
   def handle_info(%Message.PeerLeft{peer: peer}, state) do
     Membrane.Logger.info("Peer #{inspect(peer.id)} left RTC Engine")
     {_peer_channel_pid, state} = pop_in(state, [:peer_channels, peer.id])
+
+    prev_playlist_idl = hd(state.playlist_idls).playlist_idl
+
+    state =
+      update_in(state, [:playlist_idls], &Enum.reject(&1, fn x -> x[:peer_id] == peer.id end))
+
+    cond do
+      state.playlist_idls == [] ->
+        MembraneLiveWeb.Endpoint.broadcast!("event:" <> state.event_id, "playlist_playable", %{
+          playlist_idl: ""
+        })
+
+      hd(state.playlist_idls).playlist_idl != prev_playlist_idl ->
+        MembraneLiveWeb.Endpoint.broadcast!("event:" <> state.event_id, "playlist_playable", %{
+          playlist_idl: hd(state.playlist_idls).playlist_idl
+        })
+    end
```

Comment:
This is complicated and vulnerable to race conditions, so I would advise extending the HLSEndpoint message `:playlist_playable`, so it would contain also the origin of the track.